astroquery/esasky/core.py

```diff
@@ -620,15 +918,64 @@ def _get_herschel_map(self, product_url, directory_path, cache):
                     herschel_filter = self._get_herschel_filter_name(member_name)
                     tar.extract(member, directory_path)
                     observation[herschel_filter] = fits.open(
-                        directory_path +
-                        member.name)
-                    i += 1
+                        directory_path + member.name
+                    )
+                i += 1
         os.remove(tar_file.name)
         return observation
 
+    def _get_herschel_spectra(self, product_url, directory_path, cache):
+        spectra = dict()
+        tar_file = tempfile.NamedTemporaryFile(delete=False)
```

Comment:
this whole "open a file", "read from it", "close file", "delete file" could be done with `BytesIO` instead, no?  I think that would be better:
```python
from io import BytesIO
with tarfile.open(fileobj=BytesIO(response.content), 'r') as tar:
   ...
```